py4cast/lightning.py

```diff
@@ -674,10 +677,41 @@ def _next_x(
             for idx in range(batch.num_input_steps)
         ]
 
+        mask_list = []
+
+        # create a mask that corresponds to the union of the nans in the input and the forcings
+        if self.mask_on_nan:
+            combined_mask = torch.zeros_like(inputs[0][:, :, :, 0], dtype=torch.bool)
+
+            # Combine masks for inputs
+            for input in inputs:
+                mask = torch.isnan(input)
+                for i in range(mask.shape[-1]):
+                    combined_mask = (
+                        combined_mask | mask[:, :, :, i]
+                    )  # Union of size masks (batch, lat, lon)
+
+            # Combine masks for forcing
+            mask = torch.isnan(forcing.tensor)
+            for i in range(mask.shape[-1]):
+                combined_mask = (
+                    combined_mask | mask[:, :, :, i]
+                )  # Union of size masks (batch, lat, lon)
+
+            mask_list.append(
+                ~combined_mask.unsqueeze(-1)
+            )  # shape [(batch, lat, lon, param)]
+
+            # replace nan by 0 in inputs
+            inputs = [torch.nan_to_num(input) for input in inputs]
```

Comment:
Explicit default value.
```suggestion
            inputs = [torch.nan_to_num(input, nan=0) for input in inputs]
```